plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -126,7 +126,7 @@ public void testTopNPushdown()
                 .isFullyPushedDown();
 
         // TopN over limit
-        assertThat(query("SELECT orderkey, totalprice FROM (SELECT orderkey, totalprice FROM orders LIMIT 20) ORDER BY totalprice ASC LIMIT 5"))
+        assertThat(query("SELECT orderkey, totalprice FROM (SELECT orderkey, totalprice FROM orders LIMIT 15000) ORDER BY totalprice ASC LIMIT 5"))
```

Comment:
Or I can use a smaller table (but it might not catch TopN incorrectness in case of connectors that generate > 1 split).

`SELECT nationkey, regionkey FROM (SELECT nationkey, regionkey FROM nation LIMIT 25) ORDER BY regionkey LIMIT 5`

Reply:
15000 is fine. 

Side question - not related directly to this PR - do we ensure (where it is applicable) that we actually have more than one split for tests where it matters?

Reply:
nit: you can add comment that 15000 is important so tests is deterministic.

Reply:
I'll add the comment.

No we do not validate if the number of splits are as expected - let me see if there's an easy and non-intrusive way to add such a check. It would be useful because in smoke tests we use small tables which may or may not cause multiple splits to be generated.